### DISCUSSION TITLE
can custom validators also be used for custom formatting?

### DISCUSSION THREAD

**Discussion by ghost** [netbox-community/netbox#17118] [Q&A]
Let use the 'mac address' field as an example. If I input a MAC address for an interface in netbox, I can input the mac in a number of different ways.

- xx:xx:xx:xx:xx:xx
- xxxxxxxxxxxx
- xx xx xx xx xx xx
- xx-xx-xx-xx-xx-xx
- xxxx.xxxx.xxxx

IIRC those are all the accepted input methods for a MAC address. But once you click save, NetBox automatically converts any of those inputs to a consistent value. It will always convert the MAC to the following format: XX:XX:XX:XX:XX:XX

Is there a way to accomplish this same thing with custom fields and field validation? If I want to have a custom field that is supposed to be phone numbers. I can do a regex for north american 10 digit numbers that would look like this:
`^\(?\d{3}\)?[-\s]?\d{3}[-\s]?\d{4}$`

Which according to the regex documentation I have, would accept the following: 
> Examples like 123-456-7890, (123) 456-7890, and 123 456 7890 would all be accepted because the special characters are optional.

But regardless of how the 10 digit number was inputted (with or without dashes, with or without parenthesis for the area code), how would I insure that once the value is saved, that NetBox would always **display** the value consistently? 
IE: (xxx) xxx-xxxx

**Comment by chris240189** [comment]
Have you tried using an evenrule that triggers on create/update of the object the custom field is attached to?
You can run a custom script as an action of that trigger in which you parse the data with a regex and reformat it to a consistent format of your choosing.

**Comment by candlerb** [comment]
I have seen an example of a custom validator being used to fill in the default value for an empty field, so I don't see why it couldn't be used to modify the value of an existing field.
